fenix/app/src/main/java/org/mozilla/fenix/debugsettings/ui/FenixOverlay.kt

```diff
@@ -22,7 +25,22 @@ import org.mozilla.fenix.theme.Theme
 @Composable
 fun FenixOverlay() {
     val navController = rememberNavController()
-    val debugDrawerStore = remember { DebugDrawerStore() }
+    val coroutineScope = rememberCoroutineScope()
+    val debugDrawerStore = remember {
+        DebugDrawerStore(
+            middlewares = listOf(
+                DebugDrawerNavigationMiddleware(
+                    navController = navController,
+                    scope = coroutineScope,
+                ),
+            ),
+        )
+    }
+    val debugDrawerDestinations = remember {
+        DebugDrawerRoute.toDebugDrawerDestinations(
+            debugDrawerStore = debugDrawerStore,
+        )
+    }
```

Comment:
Since all of this is pure Kotlin, the preview is fully functional!!

Reply:
I guess the dream did not die after all. This is honestly kinda wild